Cython/Utility/ExtensionTypes.c

```diff
@@ -663,7 +663,12 @@ static int __Pyx__SetItemOnTypeDict(PyTypeObject *tp, PyObject *k, PyObject *v)
 #else
     result = PyDict_SetItem(tp->tp_dict, k, v);
 #endif
-    if (likely(!result)) PyType_Modified(tp);
+    if (likely(!result)) {
+        PyType_Modified(tp);
+        if (PyObject_HasAttrString(v, "__set_name__")) {
+            PyObject_CallMethod(v, "__set_name__", "OO", (PyObject *) tp, k);
```

Comment:
```suggestion
        if (PyObject_HasAttr(v, PYIDENT("__set_name__"))) {
            PyObject_CallMethodObjArgs(v, PYIDENT("__set_name__"),  (PyObject *) tp, k, NULL);
```

Fairly minor, but I think this should be slightly faster - partly because it only creates the `"__set_name__"` string once at module startup and partly because the function call should be quicker.

(`PYIDENT` is a Cython trick when processing utility code to substitute out reused strings).

With that said, this isn't really performance sensitive so doesn't matter too much.

Reply:
Thank you @da-woods. I am not expert in C API so any advice is highly appreciated.